src/tools/pub/publishing.md

```diff
@@ -319,13 +319,59 @@ pub.dev updates the package listing to display
 `2.0.0` as the stable version of the package.
 
 If all of the conditions at the beginning of this section are true,
-then you can ignore the following warning from `dart pub publish`: 
+then you can ignore the following warning from `dart pub publish`:
 
    *"Packages with an SDK constraint on a pre-release of the Dart SDK should
    themselves be published as a pre-release version. If this package needs Dart
    version 2.12.0-0, consider publishing the package as a pre-release
    instead."*
 
+
+## Retracting a package version {#retract}
+
+To prevent new package consumers from adopting a recently
+published version of your package, you can retract that package version
+within 7 days of publication.
+The retracted version can be restored again within 7 days of retraction.
+
+A retracted package version isn't deleted. It appears in the version
+listing of the package on pub.dev in the **Retracted versions** section. Also, the
+detailed view of that package version has a **RETRACTED** badge.
+
+Before retracting a package,
+consider publishing a new version instead.
+Retracting a package causes churn and can have a negative impact on package users.
```

Comment:
Maybe it's just me, but seeing "churn" here is a bit confusing and doesn't seem to add much. I'm not sure if it's necessary to have this portion of the statement or it could be clarified further. I'm fine with no changes here as well!

Reply:
Thanks for reading this through, good to hear that the description makes sense overall. 

I'll give it some thought to see if I can find a more clear phrasing instead of "churn", otherwise I might keep it as is. Thanks again.